tools/Azure.Mcp.Tools.Speech/src/Commands/BaseSpeechCommand.cs

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Diagnostics.CodeAnalysis;
+using Azure.Mcp.Core.Commands;
+using Azure.Mcp.Core.Commands.Subscription;
+using Azure.Mcp.Core.Extensions;
+using Azure.Mcp.Core.Models.Option;
+using Azure.Mcp.Tools.Speech.Options;
+
+namespace Azure.Mcp.Tools.Speech.Commands;
+
+public abstract class BaseSpeechCommand<
+    [DynamicallyAccessedMembers(TrimAnnotations.CommandAnnotations)] T>
+    : SubscriptionCommand<T>
+    where T : BaseSpeechOptions, new()
+{
+    protected override void RegisterOptions(Command command)
+    {
+        base.RegisterOptions(command);
+        var endpointOption = SpeechOptionDefinitions.Endpoint.AsRequired();
+        command.Options.Add(endpointOption);
+        command.Validators.Add(commandResult =>
+        {
+            // Validate endpoint option
+            var endpointValue = commandResult.GetValueOrDefault(endpointOption);
+            if (string.IsNullOrEmpty(endpointValue))
+            {
+                commandResult.AddError($"Missing required option: --{endpointOption.Name}");
+                return;
+            }
```

Comment:
No need to check this, options marked as required will be validated for existing centrally.